provider/openstack/firewaller.go

```diff
@@ -21,6 +21,11 @@ import (
 	"github.com/juju/juju/network"
 )
 
+var (
+	validUUID                = `[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}`
+	ExtractGroupControllerRe = regexp.MustCompile(`^(?P<prefix>juju-)(?P<controllerUUID>` + validUUID + `)(?P<suffix>-.*)$`)
```

Comment:
Please don't export the regexp itself. The pattern would be OK exported as a constant, but this thing is mutable.